src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -61,106 +56,77 @@ public static void placeConstructionTape(@NotNull final WorkOrderBuildDecoration
+                );
             }
-            newY = checkIfPlaceable(x1, y, z1, world);
-            final BlockPos corner1 = new BlockPos(x1, newY, z1);
-            newY = checkIfPlaceable(x1, y, z3, world);
-            final BlockPos corner2 = new BlockPos(x1, newY, z3);
-            newY = checkIfPlaceable(x3, y, z1, world);
-            final BlockPos corner3 = new BlockPos(x3, newY, z1);
-            newY = checkIfPlaceable(x3, y, z3, world);
-            final BlockPos corner4 = new BlockPos(x3, newY, z3);
-            world.setBlockState(corner1, BlockConstructionTape.getOptimalStateForPlacement(constructionTape.with(FACING, Direction.SOUTH), world, corner1));
-            world.setBlockState(corner2, BlockConstructionTape.getOptimalStateForPlacement(constructionTape.with(FACING, Direction.EAST), world, corner2));
-            world.setBlockState(corner3, BlockConstructionTape.getOptimalStateForPlacement(constructionTape.with(FACING, Direction.WEST), world, corner3));
-            world.setBlockState(corner4, BlockConstructionTape.getOptimalStateForPlacement(constructionTape.with(FACING, Direction.NORTH), world, corner4));
+
+            place = place.south().east();
         }
+
+        working = firstValidPosition(new BlockPos(X+W, Y, Z+H), world);
+        world.setBlockState(working,
+                BlockConstructionTape.getPlacementState(constructionTape.with(CORNER, true), world, working, Direction.WEST)
+        );
     }
 
     /**
-     * Check if a block is placeable and return new Y position.
+     * Find and return the highest position that is directly above a non-replaceable block.
      *
-     * @param x     Block X position.
-     * @param y     Block Y position.
-     * @param z     Block Z position.
+     * @param target the target position for the block
      * @param world the world.
-     * @return The new Y position.
+     * @return The new block position.
      */
-
-    public static int checkIfPlaceable(@NotNull final int x, @NotNull final int y, @NotNull final int z, @NotNull final World world)
+    public static BlockPos firstValidPosition(@NotNull BlockPos target, @NotNull World world)
     {
-        BlockPos target = new BlockPos(x,y,z);
         final Chunk chunk = world.getChunkAt(target);
 
-        target = new BlockPos(x, chunk.getTopFilledSegment() + 16, z);
-        while(world.getBlockState(target).getMaterial().isReplaceable())
+        target = new BlockPos(target.getX(), chunk.getTopFilledSegment() + 16, target.getZ());
+                
+        while (target.getY() > 0)
         {
             target = target.down();
-            if (target.getY() == 0)
-            {
+            
+            if (!world.getBlockState(target).getMaterial().isReplaceable() 
+             && !(world.getBlockState(target).getBlock() instanceof LeavesBlock)
```

Comment:
leaves block only when not persistent, or I would say no to all

Reply:
Not quite sure what you are getting at. The point of that line is so that overhanging trees are ignored. What do you mean?

Reply:
There is a property on leaves if they are placed manually by the player or builder. Those should not be replaced.

Reply:
I would say no to all, unless you can find air block under a row of leaves only

Reply:
Now does a scan for air (or something properly replaceable) below.